### DISCUSSION TITLE
nnunetv2 dataloader_train.num_processes

### DISCUSSION THREAD

**Discussion by YazdanSalimi** [Project-MONAI/tutorials#1492] [Q&A]
Thank you again for the great project.
I am using nnunetv2runner in monai. I am using --n_proc=32 for plan_and_process and it is working fine for preprocessing. 
However, even when I add --n_proc to the train function after starting the training the model uses 12 cores for dataloader and in the debug.json file the parameters are like this.
"dataloader_train.num_processes": "12",
"dataloader_val.num_processes": "6",

Could you please help me to increase these numbers?
Thank you.

**Comment by KumoLiu** [comment]
Hi @YazdanSalimi,
We follow the configuration of nnunetv2 inside `nnUNetV2Runner`; if you want to change `num_processes `, perhaps you could add `nnUNet_def_n_proc` in os.environ.
https://github.com/Project-MONAI/MONAI/blob/2daabf917a1e6afbc738d22db602f4859672a90d/monai/apps/nnunet/nnunetv2_runner.py#L207-L209

https://github.com/MIC-DKFZ/nnUNet/blob/5af388417c36b566ea029cc8b4e057d75a8ed422/nnunetv2/configuration.py#L5

Hope it helps, thanks!

**Comment by YazdanSalimi** [comment]
Thank you for your response and great project
